zerver/migrations/0565_mark_narrow_to_dm_with_welcome_bot_new_user_as_read.py

```diff
@@ -0,0 +1,70 @@
+# Generated by Django 5.0.6 on 2024-07-25 13:24
+
+from django.db import connection, migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.utils.timezone import now as timezone_now
+from psycopg2.sql import SQL
+
+
+def mark_narrow_to_dm_with_welcome_bot_new_user_as_read(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    with connection.cursor() as cursor:
+        cursor.execute(
+            SQL(
+                "SELECT MAX(id) FROM zerver_userprofile WHERE is_bot = False AND is_mirror_dummy = False;"
+            )
+        )
+        (max_id,) = cursor.fetchone()
+
+    if max_id is None:
+        return
+
+    BATCH_SIZE = 10000
+    lower_id_bound = 0
+    timestamp_value = timezone_now()
+    while lower_id_bound < max_id:
+        upper_id_bound = min(lower_id_bound + BATCH_SIZE, max_id)
+        with connection.cursor() as cursor:
+            query = SQL("""
+                INSERT INTO zerver_onboardingstep (user_id, onboarding_step, timestamp)
+                SELECT id, 'narrow_to_dm_with_welcome_bot_new_user', %(timestamp_value)s
+                FROM zerver_userprofile
+                WHERE is_bot = False
+                AND is_mirror_dummy = False
+                AND id > %(lower_id_bound)s AND id <= %(upper_id_bound)s;
+                """)
```

Comment:
This migration should only do it for `tutorial_status` values that are not "WAITING"; otherwise new users that haven't logged in yet will never get the welcome bot redirect.